src/SIL.XForge.Scripture/ClientApp/src/app/checking/checking/checking.component.ts

```diff
@@ -52,6 +52,17 @@ interface Summary {
   answered: number;
 }
 
+export enum QuestionFilter {
+  None,
+  CurrentUserHasAnswered,
+  CurrentUserHasNotAnswered,
+  HasAnswers,
+  NoAnswers,
+  StatusNone,
+  StatusExport,
+  StatusResolved
+}
+
```

Comment:
Numeric enums behave weirdly in TypeScript, the most surprising thing being that any number is assignable to it (even `-Infinity` and  `NaN`). It's also a little harder to debug if you log a value or are looking at it in the debugger, since you'll just see a number without a clear meaning.

Another situation where it makes a difference is if we forgot to add `[QuestionFilter.None]: () => true` to `questionFilterFunctions` lower in this class. As a numeric enum, the compilation error will be "Property '0' is missing in type ... ", whereas with strings the message becomes "Property 'none' is missing in type ...".

Here's my suggested change (you should be able to apply it within GitHub):

``` suggestion
export enum QuestionFilter {
  None = 'none',
  CurrentUserHasAnswered = 'current_user_has_answered',
  CurrentUserHasNotAnswered = 'current_user_has_not_answered',
  HasAnswers = 'has_answers',
  NoAnswers = 'no_answers',
  StatusNone = 'status_none',
  StatusExport = 'status_export',
  StatusResolved = 'status_resolved'
}
```

Reply:
I just discovered this has the undesired side effect of making entries be in the wrong order in the UI, since the template uses `let f of questionFilters | keyvalue`, and [the documentation for KeyValuePipe](https://angular.io/api/common/KeyValuePipe) says:
> The output array will be ordered by keys. By default the comparator will be by Unicode point value. You can optionally pass a compareFn if your keys are complex types.

Reply:
I'm retracting this suggestion in favor of a more complete solution (see other comments).